server/app/services/ti/TrustedIntermediaryService.java

```diff
@@ -143,21 +142,25 @@ private Form<EditTiClientInfoForm> validatePhoneNumber(Form<EditTiClientInfoForm
     if (Strings.isNullOrEmpty(phoneNumber)) {
       return form;
     }
-    // removes space, '(',')' and '-' from the phone number
-    phoneNumber = phoneNumber.replaceAll("[()-[\\s]]", "");
-    if (!phoneNumber.matches("[0-9]+")) {
-      return form.withError(FORM_FIELD_NAME_PHONE, "A phone number must contain only digits");
-    }
-    if (phoneNumber.length() != 10) {
-      return form.withError(FORM_FIELD_NAME_PHONE, "A phone number must contain only 10 digits");
-    }
-    try {
-      Phonenumber.PhoneNumber phonenumber = PHONE_NUMBER_UTIL.parse(phoneNumber, "US");
-      if (!PHONE_NUMBER_UTIL.isValidNumber(phonenumber)) {
-        return form.withError(FORM_FIELD_NAME_PHONE, "This phone number is not valid");
+    Optional<MessageKey> error =
+        PhoneValidationUtils.validatePhoneNumber(
+            Optional.of(phoneNumber), Optional.of(COUNTRY_CODE_FOR_US_REGION));
+    if (error.isPresent()) {
+      switch (error.get()) {
+        case PHONE_VALIDATION_NUMBER_REQUIRED:
+          return form.withError(FORM_FIELD_NAME_PHONE, "Phone number required");
+        case PHONE_VALIDATION_NON_NUMBER_VALUE:
+          return form.withError(FORM_FIELD_NAME_PHONE, "A phone number must contain only digits");
+        case PHONE_VALIDATION_INVALID_LENGTH:
+          return form.withError(
+              FORM_FIELD_NAME_PHONE, "A phone number must contain only 10 digits");
+        case PHONE_VALIDATION_INVALID_PHONE_NUMBER:
+          return form.withError(FORM_FIELD_NAME_PHONE, "This phone number is not valid");
+        case PHONE_VALIDATION_NUMBER_NOT_IN_COUNTRY:
+          return form.withError(FORM_FIELD_NAME_PHONE, "This phone number is not valid for US");
+        default:
+          return form.withError(FORM_FIELD_NAME_PHONE, "An unknown error occurred " + error.get());
```

Comment:
Suggestion (blocking): I'm hoping we don't have to copy-paste the validation messages here. Could I suggest a slightly different format for this, where we just use the `MessageKey` provided?

```
Messages messages = messagesApi.preferred(request);
if (error.isPresent()) {
  return form.withError(FORM_FIELD_NAME_PHONE, messages.at(error.get().getKeyName()));
}
return form;
```

This code assumes 1) that we can inject `MessagesApi` into the constructor of this class; and 2) That we have a valid `Request` object that we can use for `MessagesApi.preferred(request)`. I think we can do 1 and 2, but let me know if that's not the case!

Reply:
I have done one better. Passed the messages object to the service instead of request object. This reduces the scope of the overall change.

Reply:
Ooh yes even better than I imagined, great!